src/avram/charms/enum_extensions.cr

```diff
@@ -28,10 +28,37 @@ abstract struct Enum
       end
     end
 
+    def parse(value : Array(T))
+      SuccessfulCast(Array(T)).new value
+    end
+
+    def parse(values : Array(Int))
+      results = values.map { |i| parse(i) }
+      if results.all?(SuccessfulCast)
+        parse(results.map(&.value.as(T)))
+      else
+        FailedCast.new
+      end
+    end
+
+    # def parse(values : Array(String))
+    #   results = values.map {|s| parse(s)}
+    #   if results.all?(SuccessfulCast)
+    #     parse(results.map(&.value.as(T)))
+    #   else
+    #     FailedCast.new
+    #   end
+    # end
+
     def parse(value : T)
       SuccessfulCast.new(value)
     end
 
+    def to_db(values : Array(T))
+      encoded = values.map { |value| to_db(value) }.as(Array(String))
```

Comment:
I actually don't know why this needs to be String.... but if you look at the method below, it's also converting the enum value to a string. Maybe a `PQ` thing? :thinking: